src/generic/Map.jl

```diff
@@ -19,6 +19,10 @@ function (f::CompositeMap{D, C})(a) where {D, C}
    return f.map2(f.map1(a))::elem_type(C)
 end
 
+function preimage(f::Generic.CompositeMap{D, C}, a) where {D, C}
+   return preimage(f.map1, preimage(f.map2, a))
```

Comment:
We don't have a test for this, do we?

Reply:
Nope. But there is basically no map type in AA that defines a preimage function, so I wouldn't know how to do that

Reply:
Well there is `map_with_retraction_from_func` which constructs a `Generic.MapWithRetraction`. However, the code was untested before, so while adding a test was great, I do not mean to hold this PR hostage over it :-)

Reply:
There are also two tests in AA which seem to use `preimage` somehow:
```
test/generic/QuotientModule-test.jl:177:         @test h(preimage(h, n)) == n
test/generic/Submodule-test.jl:160:         pre = preimage(f, n)
```
But instead I looked at `test/generic/MapWithInverse-test.jl` and of course that resulted in discovering it is incomplete *sigh*:

```
julia> u = map_with_section_from_func(x -> x + 1, x -> x - 1, ZZ, ZZ)
Map with section
  from integers
  to integers

julia> u(ZZ(1))
2

julia> image(u, ZZ(1))
ERROR: MethodError: no method matching image(::AbstractAlgebra.Generic.MapWithSection{AbstractAlgebra.Integers{BigInt}, AbstractAlgebra.Integers{BigInt}}, ::BigInt)
Stacktrace:
 [1] top-level scope
   @ REPL[34]:1

julia> preimage(u, ZZ(1))
ERROR: MethodError: no method matching preimage(::AbstractAlgebra.Generic.MapWithSection{AbstractAlgebra.Integers{BigInt}, AbstractAlgebra.Integers{BigInt}}, ::BigInt)

Closest candidates are:
  preimage(::AbstractAlgebra.Generic.CompositeMap, ::Any)
   @ AbstractAlgebra ~/Projekte/OSCAR/AbstractAlgebra.jl/src/NemoStuff.jl:564

Stacktrace:
 [1] top-level scope
   @ REPL[35]:1
```

We only have this:
```
julia> preimage_map(u)
Map defined by a Julia function
  from integers
  to integers
```
I think this means we should open an issue.